static/app/components/group/groupSummary.spec.tsx

```diff
@@ -86,7 +94,119 @@ describe('GroupSummary', function () {
       },
     });
 
-    render(<GroupSummary groupId={groupId} />);
+    render(
+      <GroupSummary
+        groupId={groupId}
+        groupCategory={IssueCategory.ERROR}
+        groupTitle={'Uh oh.'}
+      />
+    );
+
+    await waitFor(
+      () => {
+        expect(setupCall).toHaveBeenCalled();
+      },
+      {timeout: 5000}
+    );
+
+    expect(screen.queryByText('Issue Summary')).not.toBeInTheDocument();
+    expect(screen.queryByText('Test summary')).not.toBeInTheDocument();
+    expect(screen.queryByText('Potential Impact')).not.toBeInTheDocument();
+    expect(screen.queryByText('Test impact')).not.toBeInTheDocument();
+  });
+
+  it('does not render the group summary if not an error', async function () {
+    const groupId = '1';
+    const organizationSlug = 'org-slug';
+    MockApiClient.addMockResponse({
+      url: makeGroupSummaryQueryKey(organizationSlug, groupId)[0],
+      method: 'POST',
+      body: {
+        groupId,
+        summary: 'Test summary',
+        impact: 'Test impact',
+      },
+    });
+
+    const setupCall = MockApiClient.addMockResponse({
+      url: `/issues/${groupId}/autofix/setup/`,
+      body: {
+        genAIConsent: {ok: false},
+        integration: {ok: true},
+        githubWriteIntegration: {
+          ok: true,
+          repos: [
+            {
+              provider: 'integrations:github',
+              owner: 'getsentry',
+              name: 'sentry',
+              external_id: '123',
+            },
+          ],
+        },
+      },
+    });
+
+    render(
+      <GroupSummary
+        groupId={groupId}
+        groupCategory={IssueCategory.PERFORMANCE}
+        groupTitle={'Uh oh.'}
+      />
+    );
+
+    await waitFor(
+      () => {
+        expect(setupCall).toHaveBeenCalled();
+      },
+      {timeout: 5000}
+    );
```

Comment:
We should be disabling the api query if we can determine that we aren't going to use the result ahead of time.

Reply:
Good point, done in next commit